File: stacks/ltcd_2375.cpp
//LeetCode - problem 2375

    bool check(vector<int> &ans,string &p){
        int n=p.size();
        for(int i=0;i<n;i++){
            if(p[i]=='I' && (ans[i])>=(ans[i+1]))return false;
            else if(p[i]=='D' && (ans[i])<=(ans[i+1]))return false;
        }
        return true;
    }
    string smallestNumber(string p) {
        int n=p.size();
        vector<int> ans;
        for(int i=1;i<=n+1;i++){
            ans.push_back(i);
        }
        do{
            if(check(ans,p))break;
        }while(next_permutation(ans.begin(),ans.end()));
        
        string temp;
        for(auto it:ans)temp.push_back(it+'0');
        return temp;
    }

File: stacks/ltcd_2375.py
'''
LeetCode - problem 2375
'''

    def smallestNumber(self, pattern: str) -> str:
        stack = []
        res = ''
 
        for i in range(len(pattern) + 1):
            stack.append(i + 1)
            
            if (i == len(pattern) or pattern[i] == 'I'):
                while len(stack) > 0:
                    res += str(stack.pop())
        return res
